File: Backend/app/models/inventory_model.py
from pydantic import BaseModel

class InventoryItem(BaseModel):
    item_id: str
    item_name: str
    category: str
    quantity: int
    unit_price: float
    total_price: float
    critical_level: int
    status: str

class NewInventoryItem(BaseModel):
    item_name: str
    category: str
    unit_price: float
    critical_level: int

class StockInOutInventoryItem(BaseModel):
    item_name: str
    quantity: int

class EditInventoryItem(BaseModel):
    item_name: str
    category: str
    unit_price: float
    quantity: int
    critical_level: int

File: Prototype/lib/models/inventory_model.dart
class InventoryItem {
  final String itemID;
  final String itemName;
  final String category;
  final int quantity;
  final double unitPrice;
  final double totalPrice;
  final int criticalLvl;
  final String status;

  InventoryItem({
    required this.itemID,
    required this.itemName,
    required this.category,
    required this.quantity,
    required this.unitPrice,
    required this.totalPrice,
    required this.criticalLvl,
    required this.status,
  });


  factory InventoryItem.fromJson(Map<String, dynamic> json) {
    return InventoryItem(
      itemID: json['item_id'],
      itemName: json['item_name'],
      category: json['category'],
      quantity: json['quantity'],
      unitPrice: json['unit_price'],
      totalPrice: json['total_price'],
      criticalLvl: json['critical_level'],
      status: json['status'],
    );
  }
}
